botocore/utils.py

```diff
@@ -1254,37 +1262,30 @@ def is_valid_uri(endpoint_url):
     )
 
 
-def validate_region_name(region_name):
-    """Provided region_name must be a valid host label."""
-    if region_name is None:
-        return
-    valid_host_label = re.compile(r'^(?![0-9]+$)(?!-)[a-zA-Z0-9-]{,63}(?<!-)$')
-    valid = valid_host_label.match(region_name)
-    if not valid:
-        raise InvalidRegionError(region_name=region_name)
-
+def _is_valid_host_label(value, allow_subdomains):
+    """Evaluates whether one or more string values are valid host labels per
+    RFC 1123. If allow_subdomains is True, split on `.` and recurse with
+    it set to False.
 
-def check_dns_name(bucket_name):
```

Comment:
We may have to keep this function around just in case someone imports it in their own code. Could be as simple as:

```python
def check_dns_name(bucket_name):
    return _is_valid_host_label(bucket_name, allow_subdomains=False)
```

Reply:
once again you're right! i'll update